source/_posts/durable-entities.md

```diff
@@ -0,0 +1,129 @@
+
+There are also other properties on an Entity like:
+
+**Operation name**, this would be the name of an operation that can be performed. For example `AssignBedAsync`.
+
+**Operation Input**, These are parameters that can be provided with the operation to perform. These parameters are optional. For example the name of the person that will occupy the bed.
+
+**Scheduled time**, This optional parameter indicates the specific time of delivery. Operations can be scheduled to run in the future using this parameter.
+
+A very simple Entity looks like this:
+
+```csharp
+public class BedEntity
+{
+    public string BedNumber { get; set; }
+    public bool IsOccupied { get; set; }
+    
+    private readonly ILogger _logger;
+
+    public BedEntity(ILogger logger, string bedNumber)
+    {
+        _logger = logger;
+        BedNumber = bedNumber;
+    }
+    
+    [FunctionName(nameof(BedEntity))]
+    public static async Task HandleEntityOperation(
+        [EntityTrigger] IDurableEntityContext context,
+        ILogger logger)
+    {
+        await context.DispatchAsync<BedEntity>(logger, context.EntityKey);
+    }
+    
+    public Task<bool> IsOccupiedBedAsync()
+    {
+        var IsOccupied = BedNumber == "123";
+        return Task.FromResult(IsOccupied);
+    }
+    
+    public Task AssignBedAsync(string bedNumber)
+    {
+        IsOccupied = true;
+        return Task.CompletedTask;
+    }
+}
+```
+
+If we look at the above code you might be thinking but how and where does the call to the database take place to store the `BedEntity` after its modified? This is all abstracted by the framework. We do not need to worry about it. Both methods `IsOccupiedBedAsync` and `AssignBedAsync` change the state of the entity asynchronously and then the framework does the rest. The function `HandleEntityOperation` allows us to call operations on this entity from the client function or orchestrator.
+
+## Using the Entity
+
+Now let's have a look at a simple client function that would make use of this `BedEntity`:
+
+```csharp
+public static class BedManager
+{
+    [FunctionName(nameof(AssignBed))]
+    public static async Task<IActionResult> AssignBed(
+        [HttpTrigger(AuthorizationLevel.Function, "get")]
+        HttpRequest req,
+        [DurableClient] IDurableEntityClient durableEntityClient,
+        ILogger log)
+    {
+        var bedNumber = req.Query["bedNumber"];
+        var entityId = new EntityId(nameof(BedEntity),bedNumber);
+        
+        log.LogInformation($"Received request for bed {bedNumber}");
+        
+        var bedEntity = await durableEntityClient.ReadEntityStateAsync<BedEntity>(entityId);
+        if (bedEntity.EntityExists && bedEntity.EntityState.IsOccupied)
+        {
+            return new BadRequestObjectResult("Bed is already occupied.");
+        }
```

Comment:
AssignBedAsync heeft een parameter (?) bednumber maar die is niet nodig, als je daar een patientnaam van maakt dan kun je ook tonen hoe je een parameter doorgeeft aan je durable entity function

Reply:
Ik denk persoonlijk dat dit lekker simpel is en makkelijk te begrijpen.